perf/src/sigverify.rs

```diff
@@ -496,6 +503,45 @@ impl Deduper {
     }
 }
 
+//inplace shrink a batch of packets
+pub fn shrink_batches(batches: &mut Vec<PacketBatch>) -> usize {
+    let mut i = 0;
+    let mut ii = 0;
+    let mut j = 1;
+    let mut jj = 0;
+    let mut last = 0;
```

Comment:
hahaha.  I'll refactor

Reply:
inplace updates suck in rust.  the borrow checker can't handle it.

Reply:
https://github.com/aeyakovenko/solana/pull/333

Reply:
left some comments, seems worse tbh, I don't like the label and the unsafe, while loops and breaks are easier to grok

Reply:
Thanks!  I replied there.  The while-loops that I replaced with for-loops were just for-loops in disguise.  I'd prefer the for-loops for the lower maintenance burden

Reply:
updated to use for loops and better index names.  wdyt?  I still don't like the macro over the saturating add by hand or the label or the unsafe